drake/examples/Cars/simulateLCM.cpp

```diff
@@ -78,11 +89,22 @@ int main(int argc, char* argv[]) {
   // be reused
   DrakeJoint::FloatingBaseType floating_base_type = DrakeJoint::QUATERNION;
 
-
   auto rigid_body_sys = make_shared<RigidBodySystem>();
-  rigid_body_sys->addRobotFromFile(argv[1], floating_base_type);
-  auto const & tree = rigid_body_sys->getRigidBodyTree();
-  for (int i=2; i<argc; i++)
+
+  // The following variable, weld_to_frame, is only needed if the model is a
+  // URDF file. It is needed since URDF does not specify the location and
+  // orientation of the car's root node in the world. If the model is an SDF,
+  // weld_to_frame is ignored by the parser.
```

Comment:
I considered using `Eigen::Isometry3d` instead of two `Eigen::Vector3D` objects but decided against it because I thought it would be more consistent with the use of xyz/rpy in URDF and SDF, and initializing an `Eigen::Isometry3d()`is harder and would require the user to write code similar to that in [poseValueToTransform()](https://github.com/RobotLocomotion/drake/blob/master/drake/systems/plants/xmlUtil.cpp#L79). However, I agree having two `Vector3D` parameters is confusing and may result in the user mixing up the order, and will thus switch to `Eigen::Isometry3d` as suggested.

I agree that another valid way of looking at an API-level pose specification is as the transformation between the world coordinate frame of the existing `RigidBodyTree` and the "world" coordinate frame of a URDF or SDF. Given this perspective, I agree to add the API-level pose specification with any pose specification within the model file. I will also remove the `robot_name` parameter and instead add every robot in the file. 

Since there may be multiple robots in the file, I would like to rename the method `AddRobotsFromFile()`. 

Here is the latest proposal:

```
   /**
    * Adds the robot models specified in a file to this RigidBodySystem.
    * Note that this method only adds models whose dynamics are enabled.
    * Static models like ground planes or height maps will need to be
    * added separately by calling getRigidBodyTree()->addCollisionElement().
    *
    * @param file_name The name of the file containing the robot model(s).
    * @param floating_base_type The type of the joint used in the
    * foating virtual 6 DoF that connects the robot models to the world.
    * By default, this is DrakeJoint::QUATERNION.
    * @param model_to_world_transformation The transformation between
    * this RigidBodySystem's world coordinate frame to the model's world
    * coordinate frame.
    * @return The number of robot models added.
    */
   int AddRobotsFromFile(
      const std::string& file_name,
      const DrakeJoint::FloatingBaseType floating_base_type =
          DrakeJoint::QUATERNION,
      const Eigen::Isometry3d & model_to_world_transformation
          = Isometry3d::Identity());
```
I think `RigidBodySystem`'s API should be modified to support the addition of ground planes and height maps, but that can be discussed in a future issue / PR.